src/models/model.cpp

```diff
@@ -497,8 +514,8 @@ void Model::CreateSessionOptionsFromConfig(const Config::SessionOptions& config_
 
   // If no device gets set, default to CPU
   if (!p_device_) {
-    assert(device_type_ == DeviceType::CPU);
-    p_device_ = GetDeviceInterface(device_type_);
+    assert(device_type_ == DeviceType::CPU || device_type_ == DeviceType::DML || device_type_ == DeviceType::WEBGPU);
```

Comment:
This line throws an error when `device_type_` is set to `DML`. I believe both `DML` and `WEBGPU` should use the `CPU` `DeviceInterface` to avoid this issue.

Reply:
Already fixed in main branch